File: test-fixtures/cross-language/scala/Fields.scala
class Ledger {
  var mutableField: Int = 0
  val immutableField: Int = 1
  private var privateField: String = ""
  var sharedName: Int = 0
}

object Ledger {
  var staticField: Int = 0
}

class Archive {
  var sharedName: Int = 0
}

File: test-fixtures/cross-language/cpp/fields.cpp
struct Ledger {
public:
    int mutableField;
    const int immutableField = 1;
    static int staticField;
    int sharedName;
private:
    int privateField;
};

int Ledger::staticField = 0;

struct Archive {
    int sharedName;
};

int readLedger(const Ledger &ledger) {
    return ledger.mutableField + Ledger::staticField;
}
